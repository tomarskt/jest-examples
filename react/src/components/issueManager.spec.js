import React from 'react';
import renderer from 'react-test-renderer';
import IssueManager from './issueManager';


jest.mock('../common/fetch');
describe('Issue Manager', () => {
    it('renders correctly', async () => {
        Date.now = jest.genMockFunction().mockReturnValue(1507270330454) // 6/10/2017 5:12 pm;
        const component = renderer.create(<IssueManager />);
        /**
        › 1 snapshot written.
            console.log src\components\issueManager.js:71
                ********** render with loading=false (default) ************

            console.log src\components\issueManager.js:71
                ********** render with loading=true ************  //<---------------- if you don't create time gap, this would be the render result; jest won't wait fetch Promise to resolve

            console.log src\components\issueManager.js:71
                ********** render with loading=false (final, fetch promise resolved) ************ // <--------- this is what you really want

            Snapshot Summary
            › 1 snapshot written in 1 test suite.
         */
        //******************************************* */
        await Promise.resolve(); // must create a time gap for fetch to resolve;
        //******************************************* */
        expect(component.toJSON()).toMatchSnapshot();
    })
});