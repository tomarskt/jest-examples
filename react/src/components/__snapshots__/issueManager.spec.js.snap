// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issue Manager renders correctly 1`] = `
<div>
  <div
    id="change-repo"
  >
    <label
      htmlFor="repo-url"
    >
      repo url:
    </label>
    <input
      name="repo-url"
      onChange={[Function]}
      style={
        Object {
          "width": 600,
        }
      }
      type="text"
      value="https://api.github.com/repos/facebook/react/issues?state=all"
    />
    <button
      onClick={[Function]}
    >
      Update
    </button>
  </div>
  <div
    id="issue-container"
  >
    <div
      className="issues-wrapper"
    >
      <div
        className="issue-item issue-item-filter"
      >
        <div
          className="filter-by"
        >
          <span
            className="active"
            onClick={[Function]}
          >
            3
             All
          </span>
          <span
            className=""
            onClick={[Function]}
          >
            2
             Open
          </span>
          <span
            className=""
            onClick={[Function]}
          >
            1
             Closed
          </span>
        </div>
        <div
          className="sort-by"
        >
          <label
            htmlFor="sel_sort_order"
          >
            Sort By
          </label>
          <select
            disabled={true}
          >
            <option>
              created date
            </option>
          </select>
          <select
            name="sel_sort_order"
            onChange={[Function]}
            value="descending"
          >
            <option
              value="descending"
            >
              latest first
            </option>
            <option
              value="ascending"
            >
              oldest first
            </option>
          </select>
        </div>
      </div>
      <ul>
        <li
          className="issue-item"
        >
          <div
            className="header"
          >
            <a
              className="link title"
              href="#/issues/1111"
            >
              How to wrap events in React 16
            </a>
            <span
              className="tag-0"
            >
              Component: Core
            </span>
            <span
              className="tag-1"
            >
              Type: Discussion
            </span>
          </div>
          <div
            className="content"
          >
            <div>
              #
              1111
            </div>
            <img
              className="avatar"
              src="https://avatars2.githubusercontent.com/u/23753?v=4"
            />
            <div>
              <a
                className="link"
                href="#/users/23753"
              >
                tolmasky
              </a>
              opened 
              15 hours ago
            </div>
          </div>
        </li>
        <li
          className="issue-item"
        >
          <div
            className="header"
          >
            <a
              className="link title"
              href="#/issues/3333"
            >
              How to wrap events in React 16
            </a>
            <span
              className="tag-0"
            >
              Component: Core
            </span>
            <span
              className="tag-1"
            >
              Type: Discussion
            </span>
          </div>
          <div
            className="content"
          >
            <div>
              #
              3333
            </div>
            <img
              className="avatar"
              src="https://avatars2.githubusercontent.com/u/23753?v=4"
            />
            <div>
              <a
                className="link"
                href="#/users/1111"
              >
                bochen2014
              </a>
              opened 
              2 days ago
            </div>
          </div>
        </li>
        <li
          className="issue-item"
        >
          <div
            className="header"
          >
            <a
              className="link title"
              href="#/issues/2222"
            >
              How to wrap events in React 16
            </a>
            <span
              className="tag-0"
            >
              Component: Core
            </span>
            <span
              className="tag-1"
            >
              Type: Discussion
            </span>
          </div>
          <div
            className="content"
          >
            <div>
              #
              2222
            </div>
            <img
              className="avatar"
              src="https://avatars2.githubusercontent.com/u/23753?v=4"
            />
            <div>
              <a
                className="link"
                href="#/users/3088823"
              >
                anushreesubramani
              </a>
              opened 
              3 days ago
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
